docs/v1/P4Runtime-Spec.mdk

```diff
@@ -674,6 +674,12 @@ described as follows:
   broken, in which case the P4Runtime server quickly sets one of the slave
   controllers with the highest `election_id` as master.
 
+* The mechanism via which the controller receives the P4Runtime server details
```

Comment:
by P4Runtime server details / information, do you just mean the Forwarding Pipeline Configuration, or do you include some extra information as well?

Reply:
From my experience with ONOS and P4Runtime, I'm referring to the target specific P4 device config (netcfg.json) file that is sent to the controller. The json includes the `device_id, ip, deviceKeyId, port`.

I understand the Forwarding Pipeline Configuration is sent to the switch only after the switch responds to the controller's `StreamMessageRequest`.

Reply:
Then it is worth clarifying. Also the text mentions the P4 build machine. By definition, this is the machine that produces the Forwarding Pipeline Config and sends it to the controller. I don't know what `deviceKeyId` is, I don't think that is a P4Runtime concept.

Reply:
Sure, I will clarify that. Regarding the `deviceKeyId`, your're right, it's not a part of the P4Runtime.